### PR TITLE
Enable Xinput by default

### PR DESCRIPTION
**Pull request by kiall**
On Linux, enable Xinput support by default. The Xinput lightgun driver is the
only reliable+accurate method to use lightguns on Linux, we should enable this
by default to reduce the friction for Linux Lightgun users. This should have no
effect on users of Windows/Mac etc.

### CODE REVIEW

File: scripts/src/osd/sdl.lua [hunk 1]
```diff
@@ -170,7 +170,11 @@ newoption {
 }
 
 if not _OPTIONS["NO_USE_XINPUT"] then
-	_OPTIONS["NO_USE_XINPUT"] = "1"
+	if _OPTIONS["targetos"]=="windows" or _OPTIONS["targetos"]=="macosx" or _OPTIONS["targetos"]=="haiku" or _OPTIONS["targetos"]=="asmjs" then
```
Comment (kiall) [inline scripts/src/osd/sdl.lua]: I have to admit, this feels odd vs using a `targetos==linux` check. However, I decided it was best to exactly match the X11 enable-by-default check a few lines above.

**Comment by kiall** [comment]
Nothing in there looks like it should be triggering usage as a lightgun.. So, I just tested this myself with the same machine you are using (the Geneve 9640), with or without this change, the mouse is not captured in windowed mode on Linux for me. I tested a bunch of other release tags too and built them, with varying success.

But when is it ever that simple? After doing a `make clean`, and rebuilding without xinput, it started capturing the mouse. Something, somewhere, in the build is non deterministic... 

Anyway - That led me to this code: https://github.com/mamedev/mame/blob/25adf42/src/osd/sdl/window.cpp#L359-L388

As near as I can tell, that hack is not needed for wii style lightguns, however, I only have the Ultimarc AimTrak to test with. I don't have an actual wii lightgun to check with.

I'll push a PR to make that hack optional, and off by default. Someone with a Wii lightgun will need to validate if the hack is still needed at all.

**Comment by mizapf** [comment]
I don't know enough about the SDL and XInput stuff. What I saw is quite deterministic: When I build before this XInput change, the mouse is captured. Also, when I build with the flag set in the makefile. However (this spoilt my bisect attempt at first), I have to fully clean and remake everything (a change in the makefile seems to require a make clean).
The behavior in Windows is different, once again. There the mouse is always captured, regardless of the mouse parameter. Vas once said he'll try to clean up the whole frontend stuff. Maybe we'll once get a capture/release key, or one could do that by the OSD menu.

**Comment by kiall** [comment]
> I don't know enough about the SDL and XInput stuff. What I saw is quite deterministic: When I build before this XInput change, the mouse is captured. Also, when I build with the flag set in the makefile. However (this spoilt my bisect attempt at first), I have to fully clean and remake everything (a change in the makefile seems to require a make clean).

It's the need to clean that I was referring to as non-deterministic, setting/unsetting the flag should be enough for Make to just rebuild the needed bits, for whatever reason, it doesn't!

Anyway - PR https://github.com/mamedev/mame/pull/5056 should resolve the issue.

**Pull request merged**